shared/actions/recover-password.tsx

```diff
@@ -0,0 +1,173 @@
+import * as Saga from '../util/saga'
+import * as RPCTypes from '../constants/types/rpc-gen'
+import * as RecoverPasswordGen from '../actions/recover-password-gen'
+import * as RouteTreeGen from '../actions/route-tree-gen'
+import * as Constants from '../constants/provision'
+import * as Container from '../util/container'
+import {RPCError} from '../util/errors'
+
+const chooseDevice = (params, response) => {
+  return Saga.callUntyped(function*() {
+    const devices = (params.devices || []).map(d => Constants.rpcDeviceToDevice(d))
+
+    yield Saga.put(
+      RecoverPasswordGen.createShowDeviceListPage({
```

Comment:
`displayDeviceSelect` for consistency with other stuff?

Reply:
Ah ok. My real issue is that these actions say `display*` but their receivers don't trigger the nav. For example:
https://github.com/keybase/client/blob/e3d61efb87746b074c1f0b834236750597451f6b/shared/actions/provision.tsx#L495-L507
I think you can keep the names the same, but then there should be `chainAction`s in this file that do the nav action, rather than dispatching multiple actions at the call site to get it done.